tools/linux/uio_drivers/snd/uio_snd.mk

```diff
@@ -0,0 +1,55 @@
+#
+# Copyright 2024, UNSW
+#
+# SPDX-License-Identifier: BSD-2-Clause
+#
+# This Makefile snippet builds sound UIO driver
+#
+
+ifeq ($(strip $(VMM_TOOLS)),)
+$(error VMM_TOOLS must be specified)
+endif
+ifeq ($(strip $(SDDF)),)
+$(error SDDF must be specified)
+endif
+ifeq ($(strip $(CC_USERLEVEL)),)
+$(error CC_USERLEVEL must be specified)
+endif
+
+UIO_SND_IMAGES := uio_snd_driver
+
+CFLAGS_uio_snd_driver := -I$(SDDF)/include -I$(VMM_TOOLS)/linux/include -MD
+
+CHECK_UIO_SND_DRIVER_FLAGS_HASH:=.uio_snd_driver_cflags-$(shell echo -- $(CFLAGS_USERLEVEL) $(CFLAGS_uio_snd_driver) | shasum | sed 's/ *-//')
+
+$(CHECK_UIO_SND_DRIVER_FLAGS_HASH):
+	-rm -f .uio_snd_driver_cflags-*
+	touch $@
+
+UIO_SND_DRV_DIR := $(VMM_TOOLS)/linux/uio_drivers/snd
+
+CFILES_uio_snd_driver := main.c stream.c queue.c convert.c
+OBJECTS_uio_snd_driver := $(CFILES_uio_snd_driver:.c=.o)
+DEPENDS_uio_snd_driver := $(CFILES_uio_snd_driver:.c=.d)
+
+OBJECTS_uio_snd_driver := $(addprefix uio_snd_driver.dir/,$(OBJECTS_uio_snd_driver))
+DEPENDS_uio_snd_driver := $(addprefix uio_snd_driver.dir/,$(DEPENDS_uio_snd_driver))
+
+uio_snd_driver.dir/stamp:
+	mkdir -p uio_snd_driver.dir
+	touch $@
+
+uio_snd_driver: $(OBJECTS_uio_snd_driver) 
+	$(CC_USERLEVEL) $(CFLAGS_USERLEVEL) $(CFLAGS_uio_snd_driver) $(OBJECTS_uio_snd_driver) -o $@
+
+uio_snd_driver.dir/%.o: $(UIO_SND_DRV_DIR)/%.c uio_snd_driver.dir/stamp $(CHECK_UIO_SND_DRIVER_FLAGS_HASH)
```

Comment:
Ditch the stamp file, and just use exist-depends

Reply:
Done